crypto/crypto.go

```diff
@@ -0,0 +1,29 @@
+package crypto
+
+import (
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/crypto/sha3"
+)
+
+const (
+	HashSizeByte = 32
+	HashID       = "SHAKE128"
+)
+
+func Digest(ms ...[]byte) []byte {
+	h := sha3.NewShake128()
+	for _, m := range ms {
+		h.Write(m)
+	}
+	ret := make([]byte, HashSizeByte)
+	h.Read(ret)
+	return ret
+}
+
+func Sign(privateKey []byte, message []byte) []byte {
+	return ed25519.Sign(privateKey, message)
+}
+
+func Verify(publicKey []byte, message, sig []byte) bool {
+	return ed25519.Verify(publicKey, message, sig)
+}
```

Comment:
Right, I'm not asking you to provide test vectors for ed25519.

I'd just like to see a test that generates a keypair, then uses it sign and verify something. Just something that exercises the abstract interface so that swapping out the internals is straightforward.